### DISCUSSION TITLE
Fully implict simulation of multi-parts 1D pipe flow

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#27712] [Q&A General]
Dear all.

I want to use our 1D single-phase and compressible pipe flow simulator (i.e., mass and momentum equations) to simulate the following processes where fluid flows from one vertical pipe into 2 horizontal pipes.

How can I use the known pressure boundaries at P1, P2 and P3 to directly simulate/slove the pipe flow here?

We don't want to use the MultiApp method due to its time-consuming interations.

Thanks a lot.

B

![image](https://github.com/idaholab/moose/assets/126703042/31068ad8-08f8-49ca-8891-a829fd4d9ae6)

**Comment by GiudGiud** [comment]
Hello

You can use the thermal hydraulics module for this. I m sure there is an example of a separating pipe. Maybe if you could please have a look at the junctions tests and examples for this

If you cannot find it @joshuahansel can help

Guillaume
